src/API/Google/Ads.php

```diff
@@ -323,4 +324,77 @@ private function get_merchant_link( int $merchant_id ): MerchantCenterLink {
 
 		throw new Exception( __( 'Merchant link is not available to accept', 'google-listings-and-ads' ) );
 	}
+
+	/**
+	 * Check if the user has accepted the customer data terms for enhanced conversion tracking.
+	 * Returns false for any account that fails.
+	 *
+	 * @return boolean
+	 */
+	public function get_accepted_customer_data_terms(): ?bool {
+		$ads_id = $this->options->get_ads_id();
+
+		try {
+			$accepted_terms = $this->options->get( OptionsInterface::CUSTOMER_DATA_TERMS, null );
+
+			// Retrieve the terms acceptance data from options.
+			if ( null !== $accepted_terms ) {
+				return apply_filters( 'woocommerce_gla_ads_enhanced_conversion_customer_data_terms', boolval( $accepted_terms ) );
```

Comment:
Can we typecast the returned value from `apply_filters` to a `bool`? Otherwise it's more vulnerable to being broken with a filter if the wrong return type is returned.

Also unless we need to map multiple values to a function to convert them, then typecasting is preferred instead of calling the function `boolval`.

```suggestion
				return (bool) apply_filters( 'woocommerce_gla_ads_enhanced_conversion_customer_data_terms', (bool) $accepted_terms );
```